### ISSUE TITLE
[all] Custom memory allocators wont be of use

### ISSUE THREAD

**Issue by TokyoSU** [raysan5/raylib#2568]
Please, before submitting a new issue verify and check:

 - [X] I tested it on latest raylib version from master branch
 - [X] I checked there is no similar issue already reported
 - [X] My code has no errors or misuse of raylib

### Issue description

Can't use custom memory allocators on anything except rlgl or raudio (in standalone)
since define only have this use on compiling.

### Environment

Windows 10 x64 Pro Raylib 4.0+

### Issue Screenshot

None

### Code Example

None

**Comment by raysan5** [comment]
@TokyoSU Yes, I'm afraid that's the expected behaviour. Advance users requiring custom allocators should recompile raylib defining their own allocators. At this moment there is no callback mechanism, most users don't need it.
